apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnectionStringBuilder.cs

```diff
@@ -41,7 +43,57 @@ public sealed class SpannerConnectionStringBuilder : DbConnectionStringBuilder
         /// Credentials can be retrieved from a file or obtained interactively.
         /// See Google Cloud documentation for more information.
         /// </summary>
-        public ITokenAccess Credential { get; private set; }
+        public ITokenAccess GetCredential()
+        {
+            // Check the ctor override.
+            if (CredentialOverride != null)
+            {
+                return CredentialOverride;
+            }
+
+            //Calculate it from the CredentialFile argument in the connection string.
+            GoogleCredential credential = null;
+            string jsonFile = CredentialFile;
+            if (!string.IsNullOrEmpty(jsonFile))
```

Comment:
Is there any reason we want to completely prohibit p12 files? I know json files are preferred these days, but if a user already has a p12 file, I'd expect it to work - is there something Spanner-specific here?

Reply:
My read of GoogleCredential.FromStream was that it only supported json.  As p12 has been discouraged on pantheon, I didn't consider supporting it a high priority item.  Is there an equivalent easy way to support p12?

Reply:
I think you're right - I really thought we supported p12 files, but I can't find any sign of it now.

@chrisdunelm care to chime in?

Reply:
Just checked with Chris, and you're right, while we do support p12 it's via a different call. Let's not do it for now.